### PR TITLE
Generate enum of operation paths

### LINKED ISSUE #873: Feature request: output an enum of paths
It would be very helpful to reduce duplication and have just a little bit more type safety if openapi-typescript would generate (perhaps optionally, behind a flag) an enum or object containing the API paths for each operation, so that one wouldn't have to write them down by hand or copy them manually from the spec when needing to use the paths in code.

This would help having the API spec as a single source of truth and would eliminate the need to keep paths in sync if the spec changes.

### PR DESCRIPTION
**Pull request by berzi**
Adds the `--makePathsEnum` (`-pe`) option to generate an Enum of operation paths.

Closes #873.

### CODE REVIEW

**Comment by drwpow** [comment]
Thank you for the PR! I’m the only maintainer on this repo so reviews happen as soon as I’m able to. I’ve had some personal things keeping me from being able to look at this project for a little while. Will review now.

File: src/load.ts [hunk 1]
```diff
@@ -116,7 +118,7 @@ export default async function load(
     if (urlCache.has(schemaID)) return options.schemas; // exit early if this has already been scanned
     urlCache.add(schemaID); // add URL to cache
 
-    let contents = "";
+    let contents: string;
```
Comment (drwpow) [inline src/load.ts:121]: minor nitpick, but could this be changed back to `""` for now? it makes runtime more predictable if this initialized to a `string` prototype and has access to all of those methods. an empty string is still falsy

File: src/load.ts [hunk 2]
```diff
@@ -102,7 +104,7 @@ export default async function load(
 ): Promise<{ [url: string]: PartialSchema }> {
   const urlCache = options.urlCache || new Set<string>();
 
-  const isJSON = schema instanceof URL === false; // if this is dynamically-passed-in JSON, we’ll have to change a few things
+  const isJSON = !(schema instanceof URL); // if this is dynamically-passed-in JSON, we’ll have to change a few things
```
Comment (drwpow) [inline src/load.ts:107]: please revert this; i find this harder to read

File: src/load.ts [hunk 3]
```diff
@@ -2,10 +2,13 @@ import type { GlobalContext, Headers } from "./types.js";
 import type { Dispatcher } from "undici";
 import * as undici from "undici";
 import fs from "fs";
+import yaml from "js-yaml";
+import mime from "mime";
 import path from "path";
+import type { Dispatcher } from "undici";
+import { request } from "undici";
```
Comment (drwpow) [inline src/load.ts:9]: Remove both these `undici` lines; they’re already imported above

**Review by drwpow — CHANGES_REQUESTED** [review]
Overall this is looking good! I’m in favor of merging this, but this PR needs tests. I’d be OK with either Node.js tests, CLI tests, or both (ideally both, but one will suffice, preferably CLI as it tests both).

Other than tests, I left a few comments to revert a couple lines that shouldn’t affect anything you wrote.

Once the tests are in I’d be happy to merge

**Comment by berzi** [comment]
> Thank you for the PR! I’m the only maintainer on this repo so reviews happen as soon as I’m able to. I’ve had some personal things keeping me from being able to look at this project for a little while. Will review now.

That's alright, sorry for the insistence.

I'll work on the tests and other comments soon, thank you.

**Comment by berzi** [comment]
CLI test added. 👍

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -53,6 +53,7 @@ If you’ve added a feature or fixed a bug and need to update the generated sche
 npm run build
 # 2. run the local CLI (not the npm one!)
 ./bin/cli.js tests/v3/specs/github.yaml -o tests/v3/expected/github.ts
+# NOTE: on Windows, try running the script on WSL if getting errors
```
Comment (drwpow) [inline CONTRIBUTING.md:56]: 💯

**Review by drwpow — APPROVED** [review]
Great tests! Thank you

**Comment by drwpow** [comment]
@all-contributors please add @berzi for code, docs, test

**Comment by berzi** [comment]
@drwpow Now that the PR is merged I have a question: I ran the script using `npx openapi-typescript@latest openapi.yaml -o my/path/etc -pe`, which pulled version 5.2.0, but the paths weren't generated. Do I have to wait until a version bump to access the new feature (in which case, what's the schedule?) or am I doing something wrong?

**Comment by berzi** [comment]
Update: the above also happens with 5.3.0. Not sure how to proceed. @drwpow

**Pull request merged**